src/tools/rect-selection.ts

```diff
@@ -24,66 +23,72 @@ class RectSelection {
         const root = document.createElement('div');
         root.id = 'select-root';
 
-        const updateRect = () => {
-            if (this.dragging) {
-                const x = Math.min(this.start.x, this.end.x);
-                const y = Math.min(this.start.y, this.end.y);
-                const width = Math.abs(this.start.x - this.end.x);
-                const height = Math.abs(this.start.y - this.end.y);
-
-                rect.setAttribute('x', x.toString());
-                rect.setAttribute('y', y.toString());
-                rect.setAttribute('width', width.toString());
-                rect.setAttribute('height', height.toString());
-            }
+        let dragId: number | undefined;
 
-            this.svg.style.display = this.dragging ? 'inline' : 'none';
+        const updateRect = () => {
+            const x = Math.min(this.start.x, this.end.x);
+            const y = Math.min(this.start.y, this.end.y);
+            const width = Math.abs(this.start.x - this.end.x);
+            const height = Math.abs(this.start.y - this.end.y);
+
+            rect.setAttribute('x', x.toString());
+            rect.setAttribute('y', y.toString());
+            rect.setAttribute('width', width.toString());
+            rect.setAttribute('height', height.toString());
         };
 
-        root.oncontextmenu = (e) => {
+        root.addEventListener('contextmenu', (e) => {
             e.preventDefault();
-        };
+        });
 
-        root.onmousedown = (e) => {
-            e.preventDefault();
-            e.stopPropagation();
+        root.addEventListener('pointerdown', (e) => {
+            if (dragId === undefined && ((e instanceof MouseEvent) ? e.button === 0 : e.isPrimary)) {
```

Comment:
This should always be a `PoinerEvent`. Shouldn't you do this instead:
```
e.pointerType === 'mouse'
```

Reply:
yep that's simpler, thanks